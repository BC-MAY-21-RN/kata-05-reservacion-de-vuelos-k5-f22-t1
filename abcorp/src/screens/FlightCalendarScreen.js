import React, {useState} from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {InformationOrigin} from '../components/booking/flightOrigin/InformationOrigin';
import {BtnBack} from '../components/booking/BtnBack';
import {ButtonReserve} from '../components/booking/ButtonReserve';
import {FlightCalendar} from '../components/booking/flightOrigin/FlightCalendar';

export const FlightCalendarScreen = ({route, navigation}) => {
  const {flightOrigin, flightDataComplete} = route.params;
  const flightData = route.params;

  const [date, setDate] = useState({
    day: '',
    month: '',
    year: '',
    dateComplete: '',
  });

  const {dateComplete} = date;

  return (
    <View style={styles.container}>
      <BtnBack navigation={navigation} />
      <InformationOrigin
        flightOrigin={flightOrigin}
        flightDestiny={flightDataComplete}
      />
      <Text style={styles.title}>Select date</Text>
      <FlightCalendar date={date} setDate={setDate} />
      <ButtonReserve
        navigation={navigation}
        flightDestinyScreen={'Some'}
        flightOrigin={dateComplete}
        flightDataComplete={flightData}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 30,
    flex: 1,
    marginVertical: 20,
    justifyContent: 'space-between',
    backgroundColor: 'white',
  },
  title: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 40,
  },
});
