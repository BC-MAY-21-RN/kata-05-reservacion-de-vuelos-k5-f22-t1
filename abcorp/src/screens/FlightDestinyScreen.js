import React, {useState} from 'react';
import {View, StyleSheet} from 'react-native';
import {BtnBack} from '../components/booking/BtnBack';
import {FlightForm} from '../components/booking/FlightForm';
import {InformationOrigin} from '../components/booking/flightOrigin/InformationOrigin';

export const FlightDestinyScreen = ({navigation, route}) => {
  const [flightDestiny, setflightDestiny] = useState('');
  const {flightData} = route.params;
  return (
    <View style={styles.mainContainer}>
      <BtnBack navigation={navigation} />
      <InformationOrigin flightOrigin={flightData} />
      <FlightForm
        txt="Where will you be flying to?"
        navigation={navigation}
        flightDestinyScreen="FlightCalendarScreen"
        flightOrigin={flightDestiny}
        setflightOrigin={setflightDestiny}
        flightDataComplete={flightData}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    justifyContent: 'space-between',
    marginHorizontal: 30,
    marginVertical: 20,
  },
  titleDestiny: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 40,
  },

  selectDestiny: {
    color: 'gray',
  },

  input: {
    borderBottomWidth: 0.8,
    color: 'black',
    padding: 2,
  },

  inputColor: {
    borderBottomColor: '#5D60F0',
  },
});
