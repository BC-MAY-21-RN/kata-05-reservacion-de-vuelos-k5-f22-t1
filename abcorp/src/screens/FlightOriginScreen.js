import React, {useState} from 'react';
import {View, Text, StyleSheet, TextInput} from 'react-native';
import {BtnBack} from '../components/booking/BtnBack';
// import {ButtonReserve} from '../components/booking/ButtonReserve';
import {FlightForm} from '../components/booking/FlightForm';

export const FlightOriginScreen = ({navigation}) => {
  const [flightOrigin, setflightOrigin] = useState('');
  return (
    <View style={styles.container}>
      <BtnBack navigation={navigation} />
      <FlightForm
        txt="Where are you now?"
        navigation={navigation}
        flightDestinyScreen="FlightDestinyScreen"
        flightOrigin={flightOrigin}
        setflightOrigin={setflightOrigin}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 30,
    flex: 1,
    marginVertical: 20,
    justifyContent: 'space-between',
  },
});
