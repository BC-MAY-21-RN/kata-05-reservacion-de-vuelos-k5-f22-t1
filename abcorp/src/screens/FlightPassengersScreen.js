import React, {useState} from 'react';
import {View, Text, StyleSheet, FlatList} from 'react-native';
import {BtnBack} from '../components/booking/BtnBack';
import {ButtonReserve} from '../components/booking/ButtonReserve';
import {InformationOrigin} from '../components/booking/flightOrigin/InformationOrigin';
import {SelectPassengers} from '../components/booking/flightPassengers/SelectPassengers';
import {getMonth} from '../helpers/getMonth';

const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

export const FlightPassengersScreen = ({route, navigation}) => {
  const [numberViewed, setNumberViewed] = useState(1);

  const {flightData, flightDataComplete} = route.params;
  const [year, month, day] = flightData.split('-');
  const {flightData: flightDestiny, flightDataComplete: flightOrigin} =
    flightDataComplete;
  const monthString = getMonth(month);

  const ItemSeparator = () => <View style={styles.separator} />;

  console.log(numberViewed.length);

  return (
    <View style={styles.container}>
      <BtnBack navigation={navigation} />
      <InformationOrigin
        flightOrigin={flightOrigin}
        flightDestiny={flightDestiny}
      />
      <Text style={styles.date}>
        {monthString} {day}, {year}
      </Text>
      <Text style={styles.title}>How many passengers?</Text>
      <View style={styles.list}>
        <FlatList
          data={data}
          renderItem={({item}) => (
            <SelectPassengers
              numberViewed={numberViewed}
              setNumberViewed={setNumberViewed}
              item={item}
            />
          )}
          keyExtractor={item => item}
          ItemSeparatorComponent={ItemSeparator}
        />
      </View>
      <ButtonReserve
        navigation={navigation}
        flightDestinyScreen={'HomeScreen'}
        flightData={numberViewed ? 'complete' : ''}
        flightDataComplete={''}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 30,
    flex: 1,
    marginVertical: 20,
    justifyContent: 'space-between',
    backgroundColor: 'white',
  },
  title: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 40,
  },
  date: {
    color: 'black',
    fontWeight: 'bold',
  },
  separator: {
    borderWidth: 1,
    opacity: 0.1,
  },
  list: {
    height: 250,
  },
});
