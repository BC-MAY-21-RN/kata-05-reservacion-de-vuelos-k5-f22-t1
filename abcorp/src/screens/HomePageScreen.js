import React from 'react';
import {View, Text, FlatList, TouchableOpacity, StyleSheet} from 'react-native';
import {CardFlight} from '../components/home/CardFlight';
import {ItemSeparator} from '../components/home/ItemSeparator';
import {BtnFloat} from '../components/home/BtnFloat';
import data from '../components/home/API';

export const HomePageScreen = () => {
  //const onPress = () =>
  return (
    <View style={styles.home}>
      <Text style={styles.title}>My Flights</Text>
      <FlatList
        data={data}
        renderItem={CardFlight}
        key={item => item.id}
        ItemSeparatorComponent={() => <ItemSeparator />}
        showsVerticalScrollIndicator={false}
      />
      <BtnFloat />
    </View>
  );
};

const styles = StyleSheet.create({
  home: {
    margin: 20,
    flex: 1,
  },
  title: {
    fontSize: 28,
    color: '#5D60F0',
    fontFamily: 'Impact',
    fontWeight: 'bold',
    marginTop: 10,
    marginBottom: 20,
  },
});
