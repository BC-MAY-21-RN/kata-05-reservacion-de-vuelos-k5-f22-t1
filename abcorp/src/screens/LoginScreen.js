import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {FormLogin} from '../components/Login/FormLogin';
import {ButtonsLogin} from '../components/Login/ButtonsLogin';
import {AccountRegister} from '../components/Login/AccountRegister';
import {useForm} from '../hooks/useForm';
// import {ModalLoading} from '../components/ModalLoading';

const initialForm = {
  email: '',
  password: '',
};

export const LoginScreen = ({navigation}) => {
  const {onChange, form} = useForm(initialForm);
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Log in</Text>
      <FormLogin onChange={onChange} form={form} />
      <ButtonsLogin form={form} onChange={onChange} />
      <AccountRegister navigation={navigation} />
      {/* use this component when we use firebase */}
      {/* <ModalLoading message="Login..."/> */}
    </View>
  );
};

const styles = StyleSheet.create({
  title: {
    fontSize: 22,
    marginTop: 10,
    marginBottom: 20,
    color: '#5D60F0',
    fontFamily: 'Impact',
    fontWeight: 'bold',
  },
  container: {
    marginHorizontal: 20,
    flex: 1,
    justifyContent: 'space-between',
  },
});
