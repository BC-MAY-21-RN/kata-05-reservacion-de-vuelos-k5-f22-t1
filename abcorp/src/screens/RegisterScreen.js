import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

import {FormRegister} from '../components/FormRegister';
import {ButtonsRegister} from '../components/ButtonsRegister';
import {CheckBookRegister} from '../components/CheckBookRegister';
import {HaveAccountRegister} from '../components/HaveAccountRegister';

const initialForm = {
  firstName: '',
  email: '',
  password: '',
  checkedTerm: false,
};
import {useForm} from '../hooks/useForm';

export const RegisterScreen = () => {
  const {onChange, form} = useForm(initialForm);
  const {firstName, email, password, checkedTerm, checkedSubscribed} = form;

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Sign Up</Text>
      <FormRegister
        firstName={firstName}
        email={email}
        password={password}
        onChange={onChange}
      />
      <CheckBookRegister
        checkedTerm={checkedTerm}
        onChange={onChange}
        checkedSubscribed={checkedSubscribed}
      />
      <ButtonsRegister
        firstName={firstName}
        email={email}
        password={password}
        onChange={onChange}
        checkedTerm={checkedTerm}
      />
      <HaveAccountRegister />
    </View>
  );
};

const styles = StyleSheet.create({
  title: {
    fontSize: 22,
    marginTop: 10,
    marginBottom: 20,
    color: '#5D60F0',
    fontFamily: 'Impact',
    fontWeight: 'bold',
  },
  container: {
    marginHorizontal: 20,
  },
});
