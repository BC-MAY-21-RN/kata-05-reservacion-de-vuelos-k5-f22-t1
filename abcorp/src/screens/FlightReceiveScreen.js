import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {InformationOrigin} from '../components/booking/flightOrigin/InformationOrigin';
import {ButtonReserve} from '../components/booking/ButtonReserve';
import {getMonth} from '../helpers/getMonth';
import firestore from '@react-native-firebase/firestore';
import {useSelector} from 'react-redux';
import {HomePageScreen} from './HomePageScreen';

export const FlightReceiveScreen = ({route, navigation}) => {
  const {flightDataComplete} = route.params;
  const {uid} = useSelector(state => state.auth);
  const monthString = getMonth(flightDataComplete.month);

  const addFlights = async () => {
    try {
      await firestore()
        .collection('Flights')
        .add({
          user: uid,
          flightOrigin: flightDataComplete.flightOrigin,
          flightDestiny: flightDataComplete.flightDestiny,
          passengers: flightDataComplete.numberViewed,
          date: `${flightDataComplete.year}-${flightDataComplete.month}-${flightDataComplete.day}`,
        });
    } catch (error) {
      console.log(error);
    }
  };
  return (
    <View style={styles.container}>
      <InformationOrigin
        style={styles.information}
        flightOrigin={flightDataComplete.flightOrigin}
        flightDestiny={flightDataComplete.flightDestiny}
      />
      <View />
      <View style={styles.containerInfo}>
        <Text style={styles.date}>
          {monthString} {flightDataComplete.day}, {flightDataComplete.year}
        </Text>
        <Text style={styles.date}>
          {flightDataComplete.numberViewed} passengers
        </Text>
      </View>
      <Text style={styles.title}>Your request was received.</Text>
      <ButtonReserve
        navigation={navigation}
        txtButton="Finish"
        flightDestinyScreen={''}
        flightData={'complete'}
        flightDataComplete={''}
        addFlights={addFlights}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 30,
    flex: 1,
    marginVertical: 50,
    justifyContent: 'space-evenly',
    backgroundColor: 'white',
  },
  title: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 40,
  },
  date: {
    color: 'black',
    fontWeight: 'bold',
  },
  containerInfo: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});
