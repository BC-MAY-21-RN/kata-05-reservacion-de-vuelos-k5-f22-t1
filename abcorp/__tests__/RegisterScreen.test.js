import 'react-native';
import React from 'react';
import {RegisterScreen} from '../src/screens/RegisterScreen';
import renderer from 'react-test-renderer';


describe('Tests on the component RegisterScreen', () => { 

  test('should display correctly', () => { 
    const tree = renderer.create(<RegisterScreen />);
    expect(tree).toMatchSnapshot();
   });
 });