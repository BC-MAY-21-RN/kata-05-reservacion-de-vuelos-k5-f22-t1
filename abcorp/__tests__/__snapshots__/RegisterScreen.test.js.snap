// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on the component RegisterScreen should display correctly 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#5D60F0",
        "fontFamily": "Impact",
        "fontSize": 22,
        "fontWeight": "bold",
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
  >
    Sign Up
  </Text>
  <View>
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      First Name
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        Array [
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "color": "black",
            "marginBottom": 10,
            "marginTop": 10,
            "padding": 10,
          },
          false,
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      Email
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        Array [
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "color": "black",
            "marginBottom": 10,
            "marginTop": 10,
            "padding": 10,
          },
          false,
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      Password
    </Text>
    <TextInput
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "color": "black",
            "marginBottom": 10,
            "marginTop": 10,
            "padding": 10,
          },
          false,
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "marginBottom": 10,
        }
      }
    >
      Use 8 or more characters with a mix of letters, numbers, and symbols.
    </Text>
  </View>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderWidth": 1,
            "height": 25,
            "width": 25,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "gray",
            "marginLeft": 10,
          }
        }
      >
        I agree to the Terms and Privacy Policy.
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderWidth": 1,
            "height": 25,
            "width": 25,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "gray",
            "marginLeft": 10,
          }
        }
      >
        Subscribe for select product updated.
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.2)",
        "borderRadius": 10,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 21,
          "fontWeight": "bold",
        }
      }
    >
      Sign Up
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(0,0,0,0.3)",
        }
      }
    >
      or
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.2)",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-evenly",
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/google.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 21,
          "fontWeight": "bold",
        }
      }
    >
      Sign Up with Google
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginVertical": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(0,0,0,0.3)",
        }
      }
    >
      Already have an account?
    </Text>
    <Text
      style={
        Object {
          "color": "blue",
          "textDecorationLine": "underline",
        }
      }
    >
       Log in
    </Text>
  </View>
</View>
`;
